common/lib/xmodule/xmodule/modulestore/tests/test_assetstore.py

```diff
@@ -573,79 +573,6 @@ def test_delete_all_different_type(self, storebuilder):
 
             assert len(store.get_all_asset_metadata(course.id, 'different')) == 1
 
-    @ddt.data(*MODULESTORE_SETUPS)
-    def test_get_all_assets_with_paging(self, storebuilder):
-        """
-        Save multiple metadata in each store and retrieve it singularly, as all assets, and after deleting all.
-        """
-        with storebuilder.build() as (__, store):
-            course1 = CourseFactory.create(modulestore=store)
-            course2 = CourseFactory.create(modulestore=store)
-            self.setup_assets(course1.id, course2.id, store)
-
-            expected_sorts_by_2 = (
-                (
-                    ('displayname', ModuleStoreEnum.SortOrder.ascending),
-                    ('code.tgz', 'demo.swf', 'dog.png', 'roman_history.pdf', 'weather_patterns.bmp'),
-                    (2, 2, 1)
-                ),
-                (
-                    ('displayname', ModuleStoreEnum.SortOrder.descending),
-                    ('weather_patterns.bmp', 'roman_history.pdf', 'dog.png', 'demo.swf', 'code.tgz'),
-                    (2, 2, 1)
-                ),
-                (
-                    ('uploadDate', ModuleStoreEnum.SortOrder.ascending),
-                    ('code.tgz', 'dog.png', 'roman_history.pdf', 'weather_patterns.bmp', 'demo.swf'),
-                    (2, 2, 1)
-                ),
-                (
-                    ('uploadDate', ModuleStoreEnum.SortOrder.descending),
-                    ('demo.swf', 'weather_patterns.bmp', 'roman_history.pdf', 'dog.png', 'code.tgz'),
-                    (2, 2, 1)
-                ),
-            )
-            # First, with paging across all sorts.
-            for sort_test in expected_sorts_by_2:
-                for i in range(3):
-                    asset_page = store.get_all_asset_metadata(
-                        course2.id, 'asset', start=2 * i, maxresults=2, sort=sort_test[0]
-                    )
-                    num_expected_results = sort_test[2][i]
-                    expected_filename = sort_test[1][2 * i]
-                    assert len(asset_page) == num_expected_results
-                    assert asset_page[0].asset_id.path == expected_filename
-                    if num_expected_results == 2:
-                        expected_filename = sort_test[1][(2 * i) + 1]
-                        assert asset_page[1].asset_id.path == expected_filename
```

Comment:
Lesson learned: None of these assert provide information on *which* condition above is being tested - the core assumption is that the test result will always be repeatable and not flaky, which was wrong.